--- 
title: "Leveraging Family History in Case-Control Association Analyses of Rare Variation"
author: Claudia R. Solis-Lemus$^{1*}$, S. Taylor Fischer$^{2*}$, Cuining Liu$^3$, Elizabeth J. Leslie$^1$, David J. Cutler$^1$, Debashis Ghosh$^3$, Michael P. Epstein$^1$
site: bookdown::bookdown_site
output: 
   bookdown::pdf_book:
     toc: false
     citation_package: natbib
header-includes:
- \DeclareMathOperator{\logit}{logit}
- \usepackage[dvipsnames]{xcolor}
- \newcommand{\missing}[1]{\textcolor{red}{\textbf{#1}}}
- \usepackage{setspace}\doublespacing
- \usepackage{array,ragged2e}
- \def\HS{\rule[2cm]{0pt}{0cm}}
documentclass: article
bibliography: ordinal-gamut.bib
biblio-style: apalike
link-citations: yes
---


```{r set-options, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE, 
                      error=FALSE, echo=FALSE)

library(ggplot2)
library(qqman)
##devtools::install_github("baptiste/egg")
library(grid)
library(gridExtra)
library(egg)
```

\begin{tabular}{l}
$*$ Joint first author \\[-0.25cm]
$^1$ Department of Human Genetics, Emory University, Atlanta, GA \\[-0.25cm]
$^2$ Department of Biostatistics and Bioinformatics, Emory University, Atlanta, GA \\[-0.25cm]
$^3$ Department of Biostatistics and Informatics, University of Colorado-Denver, Aurora, CO
\end{tabular}


\begin{tabular}{l}
Address for correspondence:\\[-0.25cm]
Michael Epstein\\[-0.25cm]
Department of Human Genetics\\[-0.25cm]
Emory University, School of medicine,\\[-0.25cm]
Atlanta, GA, 30050\\[-0.25cm]
Email: mpepste@emory.edu\\[-0.25cm]
Phone: (404) 712-8289
\end{tabular}

\pagebreak

\begin{center}
\textbf{Abstract} 
\end{center}

Standard methods for case-control association studies of rare variation often treat disease outcome as a dichotomous
phenotype. However, both theoretical and experimental studies have
demonstrated that subjects with a family history of disease can be
enriched for risk variation relative to subjects without such history. 
Assuming family history information is available, this observation
motivates the idea of replacing the standard dichotomous outcome
variable in case-control studies with a more informative ordinal outcome
variable that distinguishes controls ($0$), sporadic cases ($1$), and cases with a family history ($2$), and thereby leverages the trend that we should observe increasing number of risk variants with increasing category of the ordinal variable.
Here, we expand on our previous GAMuT framework [@Broadaway2016] for rare-variant association testing of multivariate phenotypes to consider ordinal traits that incorporate family history information into disease status. We use simulated data to show that, when family history information is available, our new method outperforms standard rare-variant association methods that ignore family history. We further illustrate our method using a rare-variant study of cleft lip and palate.

\pagebreak

# Introduction

Case-control association studies are among the most influential methods to identify genetic variants that are associated with a specific disease. In these studies, individuals with the disease (cases) and individuals without the disease (controls) are genotyped, and the causal effects of each genetic variant are estimated as a function of the difference in allele frequency between cases and controls, after accounting for potential confounders.
Alternatively, cohort studies include individuals who were sampled from the population of interest. Multiple genotypes and phenotypes are collected on these individuals.

@Liu2017 showed that in some association studies it can be helpful to identify family members of cases, and that even replacing cases with first-degree relatives can allow genome-wide association studies of diseases that are not present in the cohort.
In short, they propose a study design in which family members of cases are identified, instead of cases directly (GWAX: genome-wide association study by proxy). In GWAX, proxy cases are defined as the family members of affected individuals and controls are defined as the family members of unaffected individuals. This approach was proven to be powerful in genetic studies of complex traits, especially in large population cohorts.

The idea to incorporate family history information to genetic association studies can be indeed extended beyond the work of @Liu2017. 
Even in datasets that have enough sample size of cases, and thus, there is less need to define proxy-cases, incorporating family history can only improve the power to detect genetic associations with the disease by explictly accounting for enriched causal variation which is present in cases with family history of the disease.

Here, we modify the idea in @Liu2017 by distinguishing cases with family history and cases without. That is, while the disease status in GWAX has three categories: cases, proxy-cases (controls with family history) and controls; we differentiate instead between cases with family history, cases without family history, and controls. Our method can easily be extended to the case of four categories: cases with family history, cases without family history, controls with family history and controls without family history.
We extend GAMuT [@Broadaway2016], a nonparametric association method for high-dimensional phenotypes and high-dimensional genotypes, to the newly defined categorical response. We denote our proposed methodology "Ordinal GAMuT".
Family history information will enrich for causal variation, and our approach is not restricted to a specific set of categories. Our approach can be modified to only distinguish controls with family history (proxy-cases) and without, like GWAX, or to distinguish cases with family history and without. In the present work, we focus on the later to illustrate the power gains obtained from enrichment in causal variation.

Aside from allowing for an arbitrary number of genotypes and phenotypes, ordinal GAMuT can correct for confounding covariates
such as population stratification. Just as the standard GAMuT, the new ordinal GAMuT has analytical p-values,
which are scalable to genome-wide analyses.

The structure of this paper is as follows: after introducing the ordinal GAMuT method using the kernel-distance covariance framework (KDC)
[@Gretton2008, @Szekely2007, @Kosorok2009, @Zhang2012,  @Hua2015], we present simulation work to 
show that leveraging family history information via
ordinal categorical variables can greatly improve power in
rare-variant association tests compared to standard dichotomous
modeling of disease phenotypes that ignore such information, like the burden test [@Li2008] and SKAT [@Wu2011].
Finally, we apply ordinal GAMuT using genetic data from a genome-wide study of craniofacial defects
[@Leslie2016, @Leslie2016b, @Mostowska2018].

# Materials and Methods

## Leveraging Family Information through Ordinal Phenotype

We assume a sample of $N$ subjects that are genotyped for $V$ rare variants 
in a target gene or region, so that $G_j = (G_{j,1}, G_{j,2},\dots, G_{j,V})$ 
represents the genotypes of subject $j$ at $V$ rare-variant sites in the gene of interest.
Note that $G_{j,v}$ represents the number of copies of the minor allele that 
the subject possesses at the $v^{th}$ variant. Thus, the matrix of rare-variant 
genotypes for the sample is denoted $\mathbf{G} \in \mathbb{R}^{N \times V}$.

Let $\mathbf{P}$ be an $N$-dimensional vector with disease binary 
status for $N$ subjects. That is, $\mathbf{P}_j=0$ if subject $j$ is a control, 
and $\mathbf{P}_j=1$ if subject $j$ is a case.
Assuming that family history 
information is available, 
we can transform the binary phenotype into an 
ordinal phenotype by defining the ordinal score as $\mathbf{\tilde{P}}_j$ = $0$ 
if the subject is a control, $\mathbf{\tilde{P}}_j =1$ if the subject is a case 
without family history of the disease, $\mathbf{\tilde{P}}_j =2$ if the subject 
is a case with family history of the disease. Here we will focus on the situation when there are no controls with family history in the sample. However, this ordinal approach can easily be extended to the case of four categories: controls without ($\mathbf{\tilde{P}}_j=0$) and with family history ($\mathbf{\tilde{P}}_j=1$), and cases 
without ($\mathbf{\tilde{P}}_j=2$) and with family history ($\mathbf{\tilde{P}}_j=3$). The resulting phenotype vector $\mathbf{\tilde{P}}$ is an $N$-dimensional ordinal vector with disease binary 
status adjusted for family history for the $N$ subjects. 

GAMuT tests for independence between the phenotype matrix $\mathbf{\tilde{P}}$ and $\mathbf{G}$ (the $N \times V$ matrix of multivariate 
rare-variant genotypes) by constructing an $N \times N$ phenotypic-similarity 
matrix $\mathbf{Y}$, and an $N \times N$ genotypic-similarity matrix $\mathbf{X}$. 
These similarity matrices depend on a user-selected kernel function [@Kwee2008, @Schaid2010, @Wu2010, @Wu2011]. For example, the matrix $\mathbf{Y}$ can be modeled with the projection matrix: $\mathbf{Y} = \mathbf{P} (\mathbf{P}^T\mathbf{P})^{-1}\mathbf{P}^T$. Alternatively, if $\gamma(\mathbf{P}_i,\mathbf{P}_j)$ denotes the kernel function between subjects $i$ and $j$, the linear kernel is defined as $\gamma(\mathbf{P}_i,\mathbf{P}_j)= \sum_{l=1}^L P_{i,l} P_{j,l}$.
See @Broadaway2016 for more details on other kernel functions to model pairwise similarity or dissimilarity.

After constructing the simularity matrices $\mathbf{Y}$ and $\mathbf{X}$, 
we center them as $\mathbf{Y}_c = \mathbf{HYH}$ and $\mathbf{X}_c = \mathbf{HXH}$, 
where $\mathbf{H}=(\mathbf{I}-\mathbf{11}^T/N)$ is a centering matrix 
($\mathbf{HH} = \mathbf{H}$), $\mathbf{I} \in \mathbb{R}^{N \times N}$ is an 
identity matrix, and $\mathbf{1} \in \mathbb{R}^{N \times 1}$ is a vector of ones. 
With the centered similarity matrices ($\mathbf{Y}_c, \mathbf{X}_c$), we construct 
the GAMuT test statistic as

$$
T_{GAMuT}=\frac{1}{N}trace(\mathbf{Y}_c\mathbf{X}_c).
$$
Under the null hypothesis where the two matrices are independent, 
$T_{GAMuT}$ follows the asymptotic distribution as the weighted 
sum of independent and identically distributed $\chi^2_{(1)}$ variables [@Broadaway2016]. 
We then use Davies’ method [@Davies1980] to analytically calculate the p-value of $T_{GAMuT}$.


## Adjusting for Covariates
After transforming the binary phenotype to ordinal phenotype by incorporating 
the family history information, we can account for other covariates by regressing the phenotypes on covariates 
with a 
cumulative-logit regression model, and using the residuals to construct the phenotypic similarity matrix $\mathbf{Y}$.
To illustrate this procedure, let $W$ be an ordinal response with $M$ categories, and let $P(W\leq j)$ be 
the cumulative probabilities for $j=1, \dots, M$.
The proportional odds model [@McCullagh1989] is a subclass of cumulative-logit regression models and it is defined as
$$
\logit P(W \leq j | \mathbf{x}) = \theta_j - \mathbf{\beta}^T\mathbf{x} 
$$
for $j=1,\dots,M-1$. Note that the negative sign is a convention to guarantee 
that large values of $\mathbf{\beta}^T\mathbf{x}$ increase the probability in the 
larger categories. In addition, the vector of intercepts 
$\theta = (\theta_1,\dots,\theta_{M-1})$ should satisfy 
$\theta_1 \leq \theta_2 \leq \dots \leq \theta_{M-1}$.

This model is denoted proportional odds because the ratio of the 
odds of $P(W\leq j|\mathbf{x}_1)$ and $P(W\leq j|\mathbf{x}_2)$ do not 
depend on the specific category $j$.
That is,
$$
\frac{P(W \leq j | \mathbf{x}_1)/(1-P(W \leq j | \mathbf{x}_1))}{P(W \leq j | \mathbf{x}_2)/(1-P(W \leq j | \mathbf{x}_2))} = \exp(-\mathbf{\beta}^T(\mathbf{x}_1-\mathbf{x}_2))
$$
This is also denoted a \textit{parallelism assumption} on $\mathbf{\beta}$ [@Yee2010].

Note that for an ordinal response with $j$ categories, we fit $j-1$ proportional odds models. Thus, in our particular setting, we have three categories: controls ($j=0$), cases without family history ($j=1$) and cases with family history ($j=2$), and thus, we will fit 2 proportional odds models: $\logit P(W \leq 0)$ and $\logit P(W \leq 1)$.
With these models, we estimate the multinomial response probabilities for each individual. That is, for individual $i$, we have:
$$
\mu_{i,0} = P(W_i=0),
\mu_{i,1} = P(W_i=1),
\mu_{i,2} = P(W_i=2)
$$
Thus, the matrix of fitted values (denoted $\mathbf{M}$) will be a $n \times 3$ matrix where each row sums to 1, and the $i^{th}$ row corresponds to the estimated multinomial probabilities for individual $i$: $(\hat{\mu}_{i,0},\hat{\mu}_{i,1},\hat{\mu}_{i,2})$. 
To obtain the matrix of residuals, we first transform the ordinal response into a $n \times 3$ binary matrix (denoted $\mathbf{I}_W$) where the $i^{th}$ row corresponds to the 3-dimensional vector for individual $i$ with three indicator functions, one for each category: $(I(W_i=0), I(W_i=1), I(W_i=2))$ For example, if $W_i=2$, then the binary vector in the $i^{th}$ row would be $(0,0,1)$. As a result, the matrix of residuals will be the $n \times 3$ matrix of the 
difference between the binary matrix and the matrix of estimated multinomial probabilities: $\mathbf{I}_W-\mathbf{M}$.
This matrix of residuals will then be inputted into the GAMuT framework as the multiple phenotypes. 


<!--
The popularity of proportional odds can be related to its connection to a linear regression 
model on a continuous latent response (e.g. the liability score). 
That is, the ordinal variable $Y$ is obtained from a latent continuous variable $Z$ 
by $Y=k$ if $c_{k-1} < Z \leq c_k$.
Finally, a slight modification of the ordinal regression model is the use of quantities known as \textit{continuation-ratios}: $P(Y>j)/P(Y \geq j)$. A model using continuation-ratios studies the probability of stopping at $Y=j$.
-->



## Simulations

We conducted simulations to show that ordinal GAMuT properly preserves the type I error 
and to assess the power of ordinal GAMuT relative to 
standard case-control burden [@Li2008] and SKAT [@Wu2011] tests that do not account for family history information.

For the genetic data, we simulated trios (parents and offspring) with 10,000 haplotypes of 10 kb in size using COSI [@cosi], 
a coalescent model that accounts for linkage disequilibrium (LD) pattern, local recombination rate, and population 
history for individuals of European descent. We defined rare variants as those with $MAF \leq 3\%$. 
For the power simulations, we assumed the proportion of causal variants to be $15\%$, 
with effect size for each causal variant given by $\frac{\log(C)}{4}|\log_{10}(MAF)|$ plus a Normal noise with mean $0$ and variance $0.1$. We varied $C=4,6$. This setup defines the 
effect size of any given causal variant as inversely proportional to its MAF, which implies
that very rare variants will have a larger effect size.

For the ordinal phenotype, the proband’s probability of disease depended on the sequence 
data and the disease prevalence, which we varied as $0.01$, or $0.05$, while the family 
members’ probability of disease depended on the sequence data and the conditional 
recurrence risk ratio ($\lambda=2,4,8$) [@Epstein2015]. If the proband was unaffected, 
we defined the person as a control. If the proband was affected and none of the parents were affected, we defined the person 
as case without family history. Finally, if the proband was affected and at least of the parents was affected, we defined the person as a case with family history.

For each simulated dataset, we generated an equal number of controls, cases without family history, and cases with family history. We varied this number among $N=400, 750, 1000, 1500$. 
For each simulated dataset, we applied our ordinal GAMuT method that modeled cases with and without family history separately. We also applied standard burden and SKAT tests that combined all cases together without regards to family history information. For each method, we weighted rare variants using the weighting scheme recommended by @Wu2011; $w_v=Beta(MAF_v,1,25)/Beta(0,1,25)$. 

We used the R package \texttt{VGAM} and function \texttt{vglm} to fit the 
cumulative-logit regression model with proportional-odds assumption [@Yee2010], whose residuals are then used to construct the phenotypic similarity matrix input in the GAMuT package [@Broadaway2016].



<!--
Simulation setup:
1000 simulations for power, 10,000 simulations for type I error
Sequence length (bp) = 10,000
num european haplotypes = 10,000
Rare variant threshold = 0.03
% causal variants = 0.15
Disease prevalence = 0.01, 0.05
Effect size = $C \log_{10}(MAF)$; C=2,4,8
-->

<!--
From Taylor:
We simulated one binary phenotype for each subject (disease status). Then we used this disease status information and that of their siblings/parents to generate the ordinal outcome (control, case no FH, case with FH). The burden and SKAT tests use individual (binary) disease status as the trait of interest. For our method, we regress the ordinal score (as an Ord.factor variable in R w/ 3 levels "0"<"1"<"2") on the covariates, and then extract the residuals (nx3 matrix). So although we’re technically only looking at 1 phenotype (ordinal score), our phenotype matrix has 3 columns.
-->

## Analysis of Pittsburgh Orofacial Cleft Multiethnic GWAS

Orofacial clefts (OFCs) such as cleft lip (CL), cleft palate (CP), and cleft lip with cleft palate (CLP) 
are among the most common birth defects in humans with prevalence between 1 in 500 and 
1 in 2,500 live births [@Tessier1976,@Mossey2009]. Extensive recent studies have elucidated some light in the identification of common nucleotide variants associated with orofacial clefts, such as 1p22.1, 2p24.2, 3q29, 8q24.21, 10q25.3, 12q12, 16p13.3, 17q22, 17q23, 19q13, and 20q12 [@Birnbaum2009, @Grant2009, @Beaty2010, @Mangold2009, @Wolf2015, @Leslie2016, @Leslie2016b, @Mostowska2018].
However, the role of rare genetic variation in OFCs is still underway. 

The Pittsburgh Orofacial Cleft Multiethnic GWAS [@Leslie2016, @Leslie2016b] seeks to identify genetic variants that are associated with the risk of OFCs. This dataset includes a multi-ethnic cohort with 11,727 participants from 13 countries from North, Central or South America. Asia, Europe and Africa. Most of the participants were recruited as part of genetic and phenotyping studies by the University of Pittsburgh Center for Craniofacial and Dental Genetics, and some by the University of Iowa. The study cohort includes OFC-affected probands with their family members, and controls without history of OFC. Affection status consists of cleft lip (CL) with or without palate (CL/P).

We performed standard data cleaning and quality control (see @Leslie2016). We kept only Caucasian participants, and we kept rare variants with MAF in $(0.001,0.05)$ and genotype call rate greater than $95\%$.

The final sample consisted of 1411 individuals, among which there were 835 controls, 309 cases without family history and 267 cases with family history. We did not include any covariates except for the 5 PCs (see @Leslie2016 for the details on the PCA).
We applied ordinal GAMuT using linear kernel to measure pairwise phenotypic similarity.
We also ran SKAT and burden tests, with the typical weights defined in @Wu2011. For GAMuT, we used a weighted linear kernel (with the weighting scheme in @Wu2011) to measure pairwise genotypic similarity.

<!--
### Replication sample

\missing{(unclear if we want to keep this)}
The replication analysis involved 845 patients, among which 570 were controls and 275 were non-syndromic cleft lip with or without palate (nsCL/P) [@Mostowska2018]. This dataset included variants in the region 8q22.3 (hg19 chr8:$101,600,001-106,200,000$). SNPs with SNP call rate less than $0.95$ were excluded as well as samples with sample call rate less than $0.95$. Given that we do not have family information for this dataset, we could not fit the ordinal GAMuT approach here proposed. We ran SKAT and burden test for the rare variants (MAF in $(0.001,0.05)$), and standard GWAS logistic tests on the common variants under the additive model adjusting p-values with the methodology in @Conneely2007 for correlated tests.
-->

# Results


## Type I Error Simulations

Figure \@ref(fig:qq) shows the quantile-quantile (QQ) plots of 10,000 null simulations with different subjects per group, target disease prevalence, and $\lambda$ values. We show the comparison with ordinal GAMuT, SKAT and burden test. All methods compared properly control the type I error.


```{r qq, out.width='80%', fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test, SKAT, and the ordinal GAMuT here proposed. Simulated datasets (10,000) assumed a 10kb region and rare variants defined as those with MAF <3\\%. \\textbf{Top:} 750 subjects per group,  disease prevalence of 0.01 and $\\lambda=2$. \\textbf{Middle Top:} 750 subjects per group, disease prevalence of 0.05 and $\\lambda=2$. \\textbf{Middle Bottom:} 750 subjects per group, disease prevalence of 0.01 and $\\lambda=4$. \\textbf{Bottom:} 750 subjects per group, disease prevalence of 0.05 and $\\lambda=4$.',fig.show='hold', fig.align='center'}
knitr::include_graphics(c('figures/scenario1.png','figures/scenario3-2.png','figures/scenario5.png','figures/scenario7.png'))
```

## Power Simulations

Now, we compare the power of ordinal GAMuT with SKAT and burden test (Figure \@ref(fig:power-plot)). The power was estimated by computing the proportion of p-values less than the significance level ($\alpha=5 \times 10^{-5}$ for effect sizes of 4 and 6) out of 1000 replicates per scenario and model.
For these power simulations, we use different effect sizes in figures ($C=4,6$). Columns refer to the conditional recurrence risk ratio $\lambda=2,4,8$, and rows refer to disease prevalences $0.01,0.05$ . We compare the empirical power for sample sizes of $N=400,750,1000,1500$ subjects per group. Our method (ordinal GAMuT) outperformed the burden test [@Li2008] and SKAT [@Wu2011], with power increasing as all simulation parameters (sample size, disease prevalence, recurrence risk, effect size) increased. 
Our method is more powerful given that other methods merge two clearly distinct groups: cases with and without family history, and thus, they cannot exploit the information present in the enrichment of causal variants in the cases with family history. Our ordinal approach models reality better by explicitly separating these two groups that have distinct genetic characteristics.


```{r, include=FALSE}

## RV effect size = 2
dat = read.table("../results/target-prevalence-rv2.csv", header=TRUE)
dat$method = factor(dat$method, levels=c("Ordinal GAMuT", "SKAT", "Burden Test"))
alpha <- 5e-2
dat=within(dat,prev_target<-factor(prev_target, labels=c("Prevalence 0.01", "Prevalence 0.05")))
dat2 = subset(dat,Nsibsize=="Parents Only")
p1 = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  ##guides(color=guide_legend(title="Method")) +
  guides(color=FALSE) +
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 2")

## RV effect size = 4
dat = read.table("../results/target-prevalence-rv4.csv", header=TRUE)
dat$method = factor(dat$method, levels=c("Ordinal GAMuT", "SKAT", "Burden Test"))
alpha <- 5e-5
dat=within(dat,prev_target<-factor(prev_target, labels=c("Prevalence 0.01", "Prevalence 0.05")))
dat2 = subset(dat,Nsibsize=="Parents Only")
p2 = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  ##guides(color=guide_legend(title="Method")) +
  guides(color=FALSE)+
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 4")

## RV effect size = 6
dat = read.table("../results/target-prevalence-rv6.csv", header=TRUE)
dat$method = factor(dat$method, levels=c("Ordinal GAMuT", "SKAT", "Burden Test"))
alpha <- 5e-5
dat=within(dat,prev_target<-factor(prev_target, labels=c("Prevalence 0.01", "Prevalence 0.05")))
dat2 = subset(dat,Nsibsize=="Parents Only")
p3 = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")
        ##legend.justification="left"
        ##legend.position="bottom"
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  guides(color=guide_legend(title="Method")) +
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 6")

# Extract the plot legend as a separate grob
# http://stackoverflow.com/questions/12539348/ggplot-separate-legend-and-plot
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
leg = g_legend(p3)

p3no.leg = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ##legend.position="bottom"
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  guides(color=FALSE) +
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 6")
```


```{r power-plot,fig.align='center', fig.cap='Power for gene-based testing comparing three methods: burden test (blue), SKAT (green) and ordinal GAMuT (red). We compared two disease prevalences 0.01, 0.05 (rows), different conditional recurrence risk ratio $\\lambda=2,4,8$ (columns).', fig.height=8, fig.width=7}
##p1
##p2
##p3

##grid.newpage()
##grid.draw(ggarrange(p1,p2,p3, ncol = 1))
##ggarrange(p1,p2,p3, ncol = 1)

##grid.arrange(grobs = lapply(
## list(p1, p2, p3no.leg),
## set_panel_size,
## width = unit(0.75, "in"),
## height = unit(0.99, "in")
## ),leg, ncol=2)

p11 = set_panel_size(p1, width = unit(0.6, "in"), height = unit(0.95, "in"))
p22 = set_panel_size( p2, width = unit(0.6, "in"), height = unit(0.95, "in"))
p33 = set_panel_size( p3no.leg, width = unit(0.6, "in"), height = unit(0.95, "in"))

##grid.arrange(p11,p22,p33,leg,ncol=2)
grid.arrange(p22,p33,leg,ncol=3, widths=c(2.2,2.2,1))
```

## Analysis of Pittsburgh Orofacial Cleft Multiethnic GWAS

We applied our method to a Pittsburgh Orofacial Cleft (POFC) Multiethnic GWAS [@Leslie2016], [@Leslie2016b] with 1,411 Caucasian subjects (267 cases with family history, 309 cases without family history and 835 controls) and 61,671 variants used for annotation with Bystro [@Kotlar2018]. 
We filtered rare variants with MAF $[0.001,0.05]$, and filtered genes to having minimum 4 rare variants, which resulted in 5,137 gene tests. 
We tested the association between the 5,137 genes and CL or CL/P status, adjusting for PCs for population structure. We compared our results (ordinal GAMuT) with burden test and SKAT approach. 
Neither of the methods show any p-value inflation (Fig. \@ref(fig:cleft-qq)).

None of the methods identified any genes significantly associated with CL/P. However, ordinal GAMuT 
identified one gene (GRHL2) on chromosome 8 that passes the suggestive significance threshold (Fig. \@ref(fig:cleft-manhattan)). GRHL2 is in the same gene family as GRHL3, which is a transcription factor that is mutated in syndromic forms of clefting and is associated with nonsyndromic clefting [@Leslie2016, @Leslie2016b, @Carpinelli2017, @PeyrardJanvid2014].


```{r, include=FALSE}
load("../original-cleft/analysis_cleft_pass_9/by_gene/res_with_loc.Rda")
colnames(res_with_loc) <- c("gene","p.burden","p.skat","p.gamut","chr","bp")

## significance thresholds
sugg = -log10(1e-04) ##=4
gen = -log10(0.05/nrow(res_with_loc)) ##= 5.01174 
```

```{r cleft-qq, fig.width=12, fig.height=4, fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} and the ordinal GAMuT here proposed.',fig.show='hold'}
layout(matrix(c(1,2,3),nrow=1))
qq(res_with_loc$p.gamut, main = "Ordinal GAMuT")
qq(res_with_loc$p.skat, main = "SKAT")
qq(res_with_loc$p.burden, main = "Burden")
```

<!--
```{r cleft-qq2, out.width='33%', fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} and the ordinal GAMuT here proposed.',fig.show='hold'}
knitr::include_graphics(c('figures/cleft-burden-qq.png','figures/cleft-skat-qq.png','figures/cleft-ordinal-gamut-qq.png'))
```
-->



```{r cleft-manhattan, fig.width=12, fig.height=4, fig.cap='Gene-based test on Pittsburgh Orofacial Cleft (POFC) Multiethnic GWAS using burden, SKAT and ordinal GAMuT approach. Manhattan plots for each of the three tests. Red line: genome-wide significance level ($-\\log_{10}(0.05/5137) = 5.0117$). Blue line: suggestive level ($-\\log_{10}(1 \\times 10 ^{-4}) = 4$).',fig.show='hold'}
layout(matrix(c(1,2,3),nrow=1))
manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.gamut",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL, annotatePval = 1e-3,
  suggestiveline = -log10(1e-04), genomewideline = -log10(0.05/5137),main="Ordinal GAMuT",ylim=c(0,6))

manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.skat",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL, annotatePval = 1e-3,
  suggestiveline = sugg, genomewideline = gen, main="SKAT",ylim=c(0,6))

manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.burden",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL, annotatePval = 1e-3,
  suggestiveline = sugg, genomewideline = gen, main="Burden",ylim=c(0,6))
```

<!--
```{r cleft-manhattan2, out.width='33%', fig.cap='Gene-based test on Pittsburgh Orofacial Cleft (POFC) Multiethnic GWAS using burden, SKAT and ordinal GAMuT approach. Manhattan plots for each of the three tests. Red line: genome-wide significance level ($-\\log_{10}(0.05/5137) = 5.0117$). Blue line: suggestive level ($-\\log_{10}(1e-04) = 4$).',fig.show='hold'}
knitr::include_graphics(c('figures/cleft-burden.png','figures/cleft-skat.png','figures/cleft-ordinal-gamut.png'))
```
-->

<!--
### Replication analysis

Using the dataset in @Mostowska2018, we tried to replicate the top genetic associations for CL/P status. As mentioned already, the replication dataset only included the region 8q22.3 (hg19 chr8:$101,600,001-106,200,000$) and 845 individuals (570 were controls and 275 were affected with nsCL/P). SNPs with SNP call rate less than $0.95$ were excluded as well as samples with sample call rate less than $0.95$. For this dataset, we do not have family history information (and thus, we cannot fit the ordinal GAMuT method here proposed). Instead, the phenotype used was case-control status on CL/P.

We first ran a standard GWAS analysis with plink for the common variants ($MAF>0.05$) under the additive model, but none of the common variants were identified as significantly associated with CL/P affected status. The variant `rs965660` was the top hit, with adjusted p-value of $0.064419$ (at significance level of $5\%$). We used the methodology in @Conneely2007 to adjust the p-values of correlated tests.
The Q-Q plot of the unadjusted p-values is shown in figure \@ref(fig:qqplot-repl).

```{r qqplot-repl, fig.width=3, fig.height=3, fig.cap='Q-Q plots of p-values for association tests of common variants for the replication dataset. The variants belong to the region in chromosome 8 where the ordinal GAMuT top genes were found.',fig.show='hold'}
dat = read.table("../replication/data/data_us/plink-sex.assoc.logistic", header=TRUE)
dat2 = subset(dat,TEST=="ADD") ##removing pvalues for SEX
qq(dat2$P)
```


Regarding the gene-based tests for rare variants, we fit the burden test [@Li2008] and the SKAT test [@Wu2011] on 136 annotated variants with Bystro [@Kotlar2018]. We used variants with MAF between $0.1\%$ and $5\%$, which were annotated in 8 genes.
None of these genes were found to be genome-wide significant, but YWHAZ, NCALD and RIMS2 were above the suggestive significance threshold of $0.05$. Gene GRHL2 had a p-value of 0.132 with the burden test and 0.581 with the SKAT test.
-->

# Discussion

Standard GWAS methods for case-control studies usually define a disease outcome as a dichotomous phenotype. This phenotype ignores family history of disease, even if this information is available in the dataset at hand. 
Given than cases with a family history of disease can be enriched for risk variation relative to sporadic cases, incorporating family history is expected to increase power to detect genetic variants associated with disease. 


We introduce an extension to the GAMuT method [@Broadaway2016] to incorporate family information to enhance case-control association studies. This approach converts the usual binary phenotype of case-control status into an ordinal phenotype with three levels: cases with family history, cases without family history and controls, and it allows to adjust for covariates. 
Even though we do not include controls with family history, this ordinal approach can easily be extended to the case of four categories: controls with and without family history, and cases 
with and without family history by considering an ordinal phenotype with 4 levels.
Finally, just as the standard GAMuT test, the ordinal GAMuT obtains analytic p-values from Davies' method [@Davies1980] which is computationally efficient, allowing the analysis of datasets in the genomic scale.

Simulation studies of rare variant sets showed that our ordinal GAMuT method is more powerful compared to usual gene-based tests like burden test [@Li2008] and SKAT [@Wu2011] due to the fact that subjects with family history are more enriched for rare causal variants.
Applying our method to Pittsburgh Orofacial Cleft Multiethnic GWAS [@Leslie2016, @Leslie2016b], we identified a gene (GRHL2) (not previously reported) to suggestively associate with cleft lip and palate phenotypes. 
GRHL2 is in the same gene family as GRHL3, which is a transcription factor that is mutated in syndromic forms of clefting and is associated with nonsyndromic clefting [@Leslie2016, @Leslie2016b, @Carpinelli2017, @PeyrardJanvid2014].
Burden and SKAT on these same phenotypes (figure \@ref(fig:cleft-manhattan)) failed to identify any significant or suggestive genes.

We envision two main future extensions of ordinal GAMuT: 1) to include information of more distant relatives, and 2) to use disease liability as continuous phenotype instead of a categorical phenotype.
Regarding disease liability, options for enhanced outcome variables could involve conditional means from liability-threshold models which have the potential to increase the power to detect genetic variants that are associated with disease risk.
In fact, the popularity of proportional odds can be related to its connection to a linear regression 
model on a continuous latent response (e.g. the liability score). 
That is, the ordinal variable $Y$ is obtained from a latent continuous variable $Z$ 
by $Y=k$ if $c_{k-1} < Z \leq c_k$. Thus, current ordinal GAMuT which utilizes proportional odds model has a natural extension into linear regression of the latent phenotype of disease liability.
In the liability scale, family history can then be modeled as joint liability scores with a covariance matrix defined by the heritability of the disease.

<!--
To include family information in the liability model, we will need to jointly model the liability score of family members. For example, if we have information on proband and parents, we can model the joint liability as a 3-dimensional vector with liability score of proband, father and mother. This joint liability vector is assumed to be distributed multivariate normal with mean that depends on the causal effects of genetic variants, and covariance matrix that depends on the heritability of the disease. Given the disease prevalence, we can estimate the threshold in the liability scale for case-control status. That is, an individual with liability score greater than the threshold determined by the prevalence will be considered affected (case), and an individual with liability score less than the threshold will be considered unaffected (control). Just as in the univariate case, the four categories of the ordinal phenotype (case with/without family history, and controls with/without family history) can be replaced by an estimate of the liability score obtained from the conditional means of the multivariate normal distribution. For example, an unaffected individual with affected parents will have an estimated liability score given by the conditional mean of the multivariate normal under the conditioning set that the liability of the individual is less than the prevalence-determined threshold, and the liabilities of the parents are greater than this prevalence-determined threshold.
By converting the ordinal phenotype into a continuous liability score which is jointly modelled to account for family history, this extension will profit from increased power from two angles: 1) inclusion of family history (as shown in this work) and 2) use of continuous liability phenotype instead of ordinal phenotype.
-->

Finally, ordinal GAMuT is not restricted to rare genetic variants. Similar analysis could be performed for gene-based analysis of common variation.


See section of Web Resources for the R scripts to run ordinal GAMuT.


**Acknowledgements:** Samples provided by Kaare Christensen (University of Southern Denmark), Frederic W.B. Deleyiannis (University of Colorado School of Medicine, Denver), Jacqueline T. Hecht (McGovern Medical School and School of Dentistry UT Health at Houston), George L. Wehby (University of Iowa), Seth M. Weinberg (University of Pittsburgh), Jeffrey C. Murray (University of Iowa) and Mary L. Marazita (University of Pittsburgh).
This work was supported by NIH grants GM117946 [MP,DG], HG007508 [MP], R00-DE025060 [EJL], X01-HG007485 [MLM], R01-DE016148 [MLM, SMW], U01-DE024425 [MLM], R37-DE008559 [JCM, MLM], R21-DE016930 [MLM], R01-DE012472 [MLM], R01-DE011931 [JTH], R01-DE011948 [KC], U01-DD000295 [GLW]; NIH contract to the Johns Hopkins Center for Inherited Disease Research: HHSN268201200008I.


**Web Resources: ** The URLs for software: \url{https://github.com/crsl4/ordinal-gamut} and \url{http://www.genetics.emory.edu/labs/epstein/software}.


<!--
To put appendix after references: https://stackoverflow.com/questions/41532707/include-rmd-appendix-after-references
-->

<!--
## Unused code, maybe useful for some day
# g1 = ggplotGrob(p1)
# g2 = ggplotGrob(p2)
# g3 = ggplotGrob(p3)
# grid::grid.newpage()
# ##grid::grid.draw(rbind(g1,g2, g3))
# ##gridExtra::grid.arrange(g1,g2,g3)
# library(ggpubr)
# par(mar=c(0,0,0,0))
# ggarrange(g1,g2,g3,heights = c(3,3,3),
#           ncol = 1, nrow = 3, align = "v")


# Extract the plot legend as a separate grob
# http://stackoverflow.com/questions/12539348/ggplot-separate-legend-and-plot
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
thm = theme(legend.justification="left")
widths = c(8,5)
leg = g_legend(p1)

library(gridExtra)
library(ggpubr)
library(grid)
grid.arrange(p1 + guides(color=FALSE), nullGrob(), widths=widths)
grid.arrange(p2 + guides(color=FALSE), leg, widths=widths)
grid.arrange(p2 + guides(color=FALSE), nullGrob(), widths=widths)
-->

<!--
``````{r repl-manhattan, fig.width=8, fig.height=4, fig.cap='Gene-based test on the replication dataset [@Mostowska2018] using burden, and SKAT approach. Manhattan plots for each of the two tests. Red line: genome-wide significance level ($-\\log_{10}(0.05/8) = 2.20412$). Blue line: suggestive level ($-\\log_{10}(0.05) = 1.301$).',fig.show='hold'}
load("../replication/data/data_us/by_gene/res_with_loc.Rda")
colnames(res_with_loc) <- c("gene","p.burden","p.skat","chr","bp")

## significance thresholds
sugg = -log10(0.05) ##=1.301
gen = -log10(0.05/nrow(res_with_loc)) ##= 2.20412 

layout(matrix(c(1,2),nrow=1))
manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.skat",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL,annotatePval = 0.06, annotateTop = FALSE,
  suggestiveline = sugg, genomewideline = gen, main="SKAT",ylim=c(0,6), xlim=c(9e7, 1.2e8))
manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.burden",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL,annotatePval = 0.06, annotateTop = FALSE,
  suggestiveline = sugg, genomewideline = gen, main="Burden",ylim=c(0,6), xlim=c(9e7, 1.2e8))
```


```{r repl-qq, fig.width=3.5, fig.height=3.5, fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for two methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} on xxx dataset.',fig.show='hold', fig.align="center"}
load("../replication/data/data_us/by_gene/res_with_loc.Rda")
colnames(res_with_loc) <- c("gene","p.burden","p.skat","chr","bp")
source("../scripts/qqplot-MF.r")

layout(matrix(c(1,2),nrow=1))
qqunif.plot(res_with_loc$p.skat, main="SKAT")
qqunif.plot(res_with_loc$p.burden, main = "Burden")
```

```{r qq0, out.width='33%', fig.cap="Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} and the ordinal GAMuT here proposed, with target disease prevalence of 0.01, $\\lambda=2$. Simulated datasets (10,000) assumed a 10kb region and rare variants defined as those with MAF <3\\%.",fig.show='hold'}
knitr::include_graphics(c('figures/qq-burden.pdf','figures/qq-skat.pdf','figures/qq-gamut.pdf'))
```

# (APPENDIX) Appendix {-} 

# Supplementary material

```{r supp-qq, out.width='80%', fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test, SKAT, and the ordinal GAMuT here proposed. Simulated datasets (100,000) assumed a 10kb region and rare variants defined as those with MAF <3\\%. \\textbf{Top:} 750 subjects per group, disease prevalence of 0.05 and $\\lambda=2$. \\textbf{Middle:} 1000 subjects per group, disease prevalence of 0.05 and $\\lambda=2$. \\textbf{Bottom:} 750 subjects per group, disease prevalence of 0.01 and $\\lambda=4$.',fig.show='hold', fig.align='center'}
knitr::include_graphics(c('figures/scenario3.png','figures/scenario4.png','figures/scenario5.png'))
```


## Enrichment of Causal Variants

In Figure \@ref(fig:causalvar), we show that, as expected, the average number of causal rare variants and the average probability of disease is greater for the cases with family history, followed by cases without family history, and lastly for controls. This simulated dataset comprises of 1000 controls, 1000 cases without family history, and 1000 cases with family history for three levels of conditional recurrence risk ratios (columns: $\lambda=2,4,8$) and 1 or 2 siblings as family history (rows). The effect size was set as $C=2$.

```{r, include=FALSE}
##knitr::include_graphics('figures/causalvar2.png')

idat = read.table("../results/variant-stats.csv", header=TRUE, sep=",")

# n causal variants
idat$prev_target <- factor(idat$prev_target)
plotdat <- idat[idat$measure%in%"n.causal.IP",]

p1 = ggplot(data=plotdat,aes(x=group,y=mean,fill=prev_target))+
  geom_bar(color="black",stat="identity",position = "dodge")+ggtitle("Number of causal rare variants")+
  facet_grid(Nsibsize~lambda)+labs(y="Average (of 1000 Simulations)",x="Group")+
  guides(fill = guide_legend("Target Prev"))+
  geom_errorbar(aes(ymin=mean, ymax=mean+sd), width=.2,
                position=position_dodge(.9))+ 
  geom_text(
    aes(label = round(mean,2), y = mean+sd+0.08),
    position = position_dodge(0.9),
    vjust = 0, size=2
  ) + guides(fill=FALSE) +
    theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ) + ylim(low=0, high=0.67)

# prob disease
idat$prev_target <- factor(idat$prev_target)
plotdat <- idat[idat$measure%in%"pdis.IP",]
p2 = ggplot(data=plotdat,aes(x=group,y=mean,fill=prev_target))+
  geom_bar(color="black",stat="identity",position = "dodge")+ggtitle("Probability of disease")+
  facet_grid(Nsibsize~lambda)+labs(y="Average (of 1000 Simulations)",x="Group")+
 # theme(axis.text.x=element_blank(),axis.ticks.x=element_blank(), axis.title.x=element_blank())+
  guides(fill = guide_legend("Target Prev"))+
  geom_errorbar(aes(ymin=mean, ymax=mean+sd), width=.2,
                position=position_dodge(.9))+ 
  geom_text(
    aes(label = round(mean,2), y = mean+sd+0.015),
    position = position_dodge(0.9),
    vjust = 0, size=2
  ) + ##guides(fill=FALSE) +
      theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        legend.justification="left",
        axis.line = element_line(colour = "grey")##,
        ) + ylim(low=0, high=0.17)


g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
leg = g_legend(p2)

p2noleg = p2 + guides(fill=FALSE)
```


```{r causalvar,fig.align='center', fig.cap='Average of 1000 simulations of number of causal rare variants (left) and probability of disease (right) in proband for three groups: controls, cases without family history, and cases with family history under two disease prevalences (red=0.01, blue=0.05), with one (top) or two (bottom) siblings, and three conditional recurrence risk ratios as columns.', fig.height=4, fig.width=10}
grid.arrange(p1,p2noleg,leg,ncol=3,widths=c(4,4,1))
```

-->