--- 
title: "Enhanced Case-Control Association Analyses Leveraging Family History"
author: "Claudia Solis-Lemus$^+$, Taylor Fischer$^+$, Cuining Liu, Elizabeth Leslie, David Cutler, Debashis Ghosh, Michael Epstein"
site: bookdown::bookdown_site
output: 
   bookdown::pdf_book:
     toc: false
     citation_package: natbib
header-includes:
- \DeclareMathOperator{\logit}{logit}
- \usepackage[dvipsnames]{xcolor}
- \newcommand{\missing}[1]{\textcolor{red}{\textbf{#1}}}
documentclass: article
bibliography: ordinal-gamut.bib
biblio-style: apalike
link-citations: yes
---


```{r set-options, echo=FALSE, include=FALSE}
knitr::opts_chunk$set(message=FALSE, warning=FALSE, 
                      error=FALSE, echo=FALSE)

library(ggplot2)
library(qqman)
##devtools::install_github("baptiste/egg")
library(grid)
library(gridExtra)
library(egg)
```


\noindent \textbf{Abstract.} Standard methods for case-control association studies of rare and
common variation often treat disease outcome as a dichotomous
phenotype. However, both theoretical and experimental studies have
demonstrated that cases with a family history of disease can be
enriched for risk variation relative to sporadic cases. Likewise,
controls with a family history of disease can harbor more risk
variation than controls without such history.
Assuming family-history information is available, this observation
motivates the idea of replacing the standard dichotomous outcome
variable in case-control studies with a more informative outcome
variable that distinguishes cases with and without family history, and
controls. The rationale is that subjects with family history can
be more enriched for variants of most common diseases. That is, we
expect to see more causal variants in cases with family history and
want to leverage this trend when exploring the genetic contributors of complex traits.
Here, we propose an extension to the cross-phenotype association method GAMuT [@Broadaway2016] 
which tests association among ordinal phenotype data with high-dimensional genotype data. 
The test aggregates information
from rare variants within a gene, and incorporates family history information by transforming the
case-control status into an ordinal trait with three levels: cases with family history, cases without
family history and controls. We denote this extension "Ordinal GAMuT".

# Introduction

The case–control association study is a powerful method for identifying genetic variants that influence disease risk. In a typical study, a researcher genotypes a set of individuals who have a disease (the 'cases') and a set of individuals who do not have the disease (the 'controls'). For each genetic variant, the difference in allele frequency between cases and controls can be used to estimate the causal effect of the genetic variant on the disease (assuming all potential confounders have been accounted for). While powerful, this study design requires an a priori decision about which disease is of interest, as well as substantial effort to identify matched cases and controls. An alternative approach is a cohort study in which individuals are sampled from the general population and many phenotypes (along with genotypes) are collected on each individual. An advantage of a cohort study is that the cohort can be subdivided to create case–control studies of many different diseases.

However, cohort studies are limited by the fact that unbiased sampling may not yield sufficient numbers of cases to enable powerful case–control studies. For example, even in a perfectly representative sample of 1 million people, one expects only 10,000 cases of a disease like schizophrenia (with a population prevalence of 1%). Further, participants in a cohort study are rarely a fully representative sample of a given population; a disease may also be rare in a cohort for the simple reason that the sampled population does not include the demographic group where the disease is most prevalent. For example, the UK Biobank (an ongoing and widely available cohort study) sampled individuals in the age range of 40–69 years (at the time of recruitment)1. By definition, this cohort does not include individuals with lethal childhood diseases, and at present there are only a handful of individuals with Alzheimer's disease or other late-onset diseases. Similarly, cohort studies that focus on individuals of a single sex (like the Nurses Health Study) have little power to study diseases that are more common in the other sex. Other sampling approaches, like cohorts made from customers of consumer genomics companies (for example, see ref. 2 or our crowdsourcing website, https://dna.land), have analogous limitations. More generally, the number of cases of a given disease present in a cohort will be a function of aspects of the disease (with rarely occurring or rapidly lethal diseases being rarer) and aspects of the sampling.

In this paper, we consider a study design where the researcher identifies family members of cases, rather than cases themselves (as the cases may be difficult or impossible to contact). This design is analogous to genomic selection in animal breeding, where the phenotypes of relatives are regularly incorporated into calculating breeding values3,4,5. In human studies, this design has been discussed in the context of family-based association studies6,7 and is popular in studies of longevity (where 'cases' are long-lived individuals8,9,10,11) but has not been widely used in other situations. The approach can be thought of as taking pedigree-based association methods that allow for missing genotype data12,13,14 to an extreme where no cases have been genotyped or phenotyped by the researcher.

As a motivating example for this type of design, consider Alzheimer's disease. As of 25 March 2016, there were 55 cases of Alzheimer's disease listed among the approximately 500,000 individuals in the UK Biobank. However, over 60,000 participants noted that one or both of their parents was/is affected with the disease. An individual with a single affected parent can be thought to have one chromosome sampled from a population of 'cases' and one sampled from a population of 'controls'. If the allele frequency (in the standard case–control setting) of some variant that increases risk of a disease is fA in cases and fU in controls, then the allele frequency in individuals with a single affected parent is (fU + fA)/2. This motivates a proxy case–control association study where 'proxy cases' are the relatives of affected individuals and controls are the relatives of unaffected individuals. We refer to this approach as a genome-wide association study by proxy (GWAX).

\missing{xxxxxxxx}

Aside from allowing for an arbitrary number of genotypes, ordinal GAMuT can correct for confounding covariates
such as population stratification. Just as the standard GAMuT, the new ordinal GAMuT has analytical p-values,
which are scalable to genome-wide analyses.

After introducting the ordinal GAMuT method using the kernel-distance covariance framework (KDC)
[@Gretton2008, @Szekely2007, @Kosorok2009, @Zhang2012,  @Hua2015], we present simulation work to 
show that leveraging family-history information via
ordinal categorical variables can greatly improve power in
rare-variant association tests compared to standard dichotomous
modeling of disease phenotypes that ignore such information, like the burden test [@Li2008] and SKAT [@Wu2011].
Finally, we apply ordinal GAMuT using GWAS
and NGS data from a genome-wide study of craniofacial defects
[@Leslie2016], [@Leslie2016b], [@Mostowska2018].

# Materials and Methods

## GAMuT Test of Cross-Phenotype Associations

GAMuT [@Broadaway2016] studies the association between high-dimensional phenotypes and 
high-dimensional genotypes, through a nonparametric test of independence 
between two sets of multivariate variables. 
We assume a sample of $N$ subjects that are genotyped for $V$ rare variants 
in a target gene or region. We assume these subjects are measured for $L$ phenotypes. 
For subject $j$ $(j = 1,\dots,N)$, let 
$\mathbf{P}_j = (P_{j,1}, P_{j,2},\dots, P_{j,L})$ be the $L$-dimensional 
vector of phenotypes, so that $\mathbf{P} \in \mathbb{R}^{N \times L}$ is 
the full matrix of phenotypes. Similarly, let $G_j = (G_{j,1}, G_{j,2},\dots, G_{j,V})$ 
be the genotypes of subject $j$ at $V$ rare-variant sites in the gene of interest.
Note that $G_{j,v}$ represents the number of copies of the minor allele that 
the subject possesses at the $v^{th}$ variant. Thus, the matrix of rare-variant 
genotypes for the sample is denoted $\mathbf{G} \in \mathbb{R}^{N \times V}$.


GAMuT tests for independence between $\mathbf{P}$ (the $N \times L$ matrix of 
multivariate phenotypes) and $\mathbf{G}$ (the $N \times V$ matrix of multivariate 
rare-variant genotypes) by constructing an $N \times N$ phenotypic-similarity 
matrix $\mathbf{Y}$, and an $N \times N$ genotypic-similarity matrix $\mathbf{X}$. 
See @Broadaway2016 for more details on how to model pairwise similarity or dissimilarity.

After constructing the simularity matrices $\mathbf{Y}$ and $\mathbf{X}$, 
we center them as $\mathbf{Y}_c = \mathbf{HYH}$ and $\mathbf{X}_c = \mathbf{HXH}$, 
where $\mathbf{H}=(\mathbf{I}-\mathbf{11}^T/N)$ is a centering matrix 
($\mathbf{HH} = \mathbf{H}$), $\mathbf{I} \in \mathbb{R}^{N \times N}$ is an 
identity matrix, and $\mathbf{1} \in \mathbb{R}^{N \times 1}$ is a vector of ones. 
With the centered similarity matrices ($\mathbf{Y}_c, \mathbf{X}_c$), we construct 
the GAMuT test statistic as

$$
T_{GAMuT}=\frac{1}{N}trace(\mathbf{Y}_c\mathbf{X}_c).
$$
Under the null hypothesis where the two matrices are independent, 
$T_{GAMuT}$ follows the asymptotic distribution as the weighted 
sum of independent and identically distributed $\chi^2_{(1)}$ variables [@Broadaway2016]. 
We then use Davies’ method [@Davies1980] to analytically calculate the p-value of $T_{GAMuT}$.

## Leveraging Family Information through Ordinal phenotype

First, we assume for simplicity that we only have one phenotype in our dataset ($L=1$), 
so that 
$\mathbf{P}$ is an $N$-dimensional vector with disease binary 
status for $N$ subjects. That is, $\mathbf{P}_j=0$ if subject $j$ is a control, 
and $\mathbf{P}_j=1$ if subject $j$ is a case. Assuming that family-history 
information is available, 
we can transform the binary phenotype into an 
ordinal phenotype by defining the ordinal score as $\mathbf{\tilde{P}}_j$ = $0$ 
if the subject is a control, $\mathbf{\tilde{P}}_j =1$ if the subject is a case 
without family history of the disease, $\mathbf{\tilde{P}}_j =2$ if the subject 
is a case with family history of the disease. Here we describe the case when there are no controls with family history. However, this ordinal approach can easily be extended to the case of four categories: controls with ($\mathbf{\tilde{P}}_j=1$) and without family history ($\mathbf{\tilde{P}}_j=1$), and cases 
with ($\mathbf{\tilde{P}}_j=3$) and without family history ($\mathbf{\tilde{P}}_j=2$).

Similarly to GAMuT, we construct a centered similarity matrix for the 
phenotypes ($\mathbf{Y}_c$) and a centered similarity matrix for the genotypes 
($\mathbf{X}_c$), and test for the independence 
between these two matrices [@Broadaway2016].


## Adjusting for covariates
After transforming the binary phenotype to ordinal phenotype by incorporating 
the family-history information, we account for other covariates with a 
cumulative-logit regression model.
Let $Y$ be an ordinal response with $M$ levels, and let $P(Y\leq j)$ be 
the cumulative probabilities for $j=1, \dots, M$.
The proportional odds model [@McCullagh1989] is defined as
$$
\logit P(Y \leq j | \mathbf{x}) = \theta_j - \mathbf{\beta}^T\mathbf{x} 
$$
for $j=1,\dots,M-1$. Note that the negative sign is a convention to guarantee 
that large values of $\mathbf{\beta}^T\mathbf{x}$ increase the probability in the 
larger categories. In addition, the vector of intercepts 
$\theta = (\theta_1,\dots,\theta_M-1)$ should satisfy 
$\theta_1 \leq \theta_2 \leq \dots \leq \theta_M-1$.

This model is denoted proportional odds because the ratio of the 
odds of $P(Y\leq j|\mathbf{x}_1)$ and $P(Y\leq j|\mathbf{x}_2)$ do not 
depend on the specific category $j$.
That is,
$$
\frac{P(Y \leq j | \mathbf{x}_1)/(1-P(Y \leq j | \mathbf{x}_1))}{P(Y \leq j | \mathbf{x}_2)/(1-P(Y \leq j | \mathbf{x}_2))} = \exp(-\mathbf{\beta}^T(\mathbf{x}_1-\mathbf{x}_2))
$$
This is also denoted a \textit{parallelism assumption} on $\mathbf{\beta}$ [@Yee2010].

The residuals of the proportional odds model are then inputted into the GAMuT framework as the multiple phenotypes: one set of residuals for every $j=0,1,2$ when fitting the model on $\logit P(Y \leq j | \mathbf{x})$. These residuals are not influenced by the confounding covariates anymore, and thus, can be directly tested for association with the genotypes.


<!--
The popularity of proportional odds can be related to its connection to a linear regression 
model on a continuous latent response (e.g. the liability score). 
That is, the ordinal variable $Y$ is obtained from a latent continuous variable $Z$ 
by $Y=k$ if $c_{k-1} < Z \leq c_k$.
Finally, a slight modification of the ordinal regression model is the use of quantities known as \textit{continuation-ratios}: $P(Y>j)/P(Y \geq j)$. A model using continuation-ratios studies the probability of stopping at $Y=j$.
-->




## Simulations

We conducted simulations to show that ordinal GAMuT properly preserves the type I error 
and to assess the power of ordinal GAMuT relative to approaches that do not account for 
family-history information by using a binary response, like the burden test [@Li2008] 
and SKAT [@Wu2011].

For the genetic data, we simulated 10,000 haplotypes of 10 kb in size using COSI [@cosi], 
a coalescent model that accounts for LD pattern, local recombination rate, and population 
history for individuals of European descent. We defined rare variants as those with $MAF<3\%$. 
For the power simulations, we assumed the proportion of causal variants to be $15\%$, 
with effect size given by $C|\log_{10}(MAF)|$, where we varied $C=2,4,8$. This setup defines the 
effect size of any given causal variant as inversely proportional to its MAF, which implies
that very rare variants will have a larger effect size.

For the ordinal phenotype, the proband’s probability of disease depended on the sequence 
data and the disease prevalence, which we varied as $0.01,0.05$, while the family 
members’ probability of disease depended on the sequence data and the conditional 
recurrence risk ratio ($\lambda=2,4,8$) [@Epstein2015]. If the proband was unaffected, 
we defined the person as a control. For family history, we simulated three different scenarios: 1) parents as family members, 
2) one sibling as family member, or 3) two siblings as family members. If the proband was affected, we defined the person 
as case without family history if family members were unaffected or a case with family 
history if family members were affected.
For each simulated dataset, we generated an equal number of controls, cases without family history, and cases with family history. We varied this number among $N=400, 750, 1000, 1500$. 

For each simulated dataset, we applied our ordinal GAMuT method that modeled cases with and without family history separately. We also applied standard burden and SKAT tests that combined all cases together without regards to family history information. For each method, we weighted rare variants using the weighting scheme recommended by @Wu2011; $w_v=Beta(MAF_v,1,25)/Beta(0,1,25)$. 

We used the R package \texttt{VGAM} and function \texttt{vglm} to fit the 
cumulative-logit regression model with proportional-odds assumption [@Yee2010].



<!--
Simulation setup:
1000 simulations for power, 10,000 simulations for type I error
Sequence length (bp) = 10,000
num european haplotypes = 10,000
Rare variant threshold = 0.03
% causal variants = 0.15
Disease prevalence = 0.01, 0.05
Effect size = $C \log_{10}(MAF)$; C=2,4,8
-->

<!--
From Taylor:
We simulated one binary phenotype for each subject (disease status). Then we used this disease status information and that of their siblings/parents to generate the ordinal outcome (control, case no FH, case with FH). The burden and SKAT tests use individual (binary) disease status as the trait of interest. For our method, we regress the ordinal score (as an Ord.factor variable in R w/ 3 levels "0"<"1"<"2") on the covariates, and then extract the residuals (nx3 matrix). So although we’re technically only looking at 1 phenotype (ordinal score), our phenotype matrix has 3 columns.
-->

## Analysis of Pittsburgh Orofacial Cleft Multiethnic GWAS

Orofacial clefts (OFCs) such as cleft lip (CL), cleft palate (CP), and cleft lip with cleft palate (CLP) 
are among the most common birth defects in humans with prevalence between 1 in 500 and 
1 in 2,500 live births [@Tessier1976], [@Mossey2009]. Extensive recent studies have elucidated some light in the identification of common nucleotide variants associated with orofacial clefts, such as 1p22.1, 2p24.2, 3q29, 8q24.21, 10q25.3, 12q12, 16p13.3, 17q22, 17q23, 19q13, and 20q12 [@Birnbaum2009, @Grant2009, @Beaty2010, @Mangold2009, @Wolf2015, @Leslie2016, @Leslie2016b, @Mostowska2018].
However, the role of rare genetic variation in OFCs is still underway. 

The Pittsburgh Orofacial Cleft Multiethnic GWAS [@Leslie2016, @Leslie2016b] seeks to identify genetic variants that are associated with the risk of OFCs. This dataset includes a multi-ethnic cohort with 11,727 participants from 13 countries from North, Central or South America. Asia, Europe and Africa. Most of the participants were recruited as part of genetic and phenotyping studies by the University of Pittsburgh Center for Craniofacial and Dental Genetics, and some by the University of Iowa. The study cohort includes OFC-affected probands with their family members, and controls without history of OFC. Affection status consists of cleft lip (CL) with or without palate (CL/P).

We performed standard data cleaning and quality control (see @Leslie2016). We kept only Caucasian participants, and we kept rare variants with MAF in $(0.001,0.05)$ and genotype call rate greater than $95\%$.

The final sample consisted of 1411 individuals, among which there were 835 controls, 309 cases without family history and 267 cases with family history. We did not include any covariates except for the 5 PCs (see @Leslie2016 for the details on the PCA).
We applied ordinal GAMuT using linear kernel to measure pairwise phenotypic similarity.
We also ran SKAT and burden tests. For GAMuT and SKAT tests, we used a weighted linear kernel (with the weighting scheme in @Wu2011) to measure pairwise genotypic similarity.

### Replication sample

The replication analysis involved 845 patients, among which 570 were controls and 275 were non-syndromic cleft lip ith or without palate (nsCL/P) [@Mostowska2018]. This dataset included variants in the region 8q22.3 (hg19 chr8:$101,600,001-106,200,000$). SNPs with SNP call rate less than $0.95$ were excluded as well as samples with sample call rate less than $0.95$. Given that we do not have family information for this dataset, we could not fit the ordinal GAMuT approach here proposed. We ran SKAT and burden test for the rare variants (MAF in $(0.001,0.05)$), and standard GWAS logistic tests on the common variants under the additive model adjusting p-values with the methodology in @Conneely2008 for correlated tests.

# Results

## Enrichment of causal variants

In figure \@ref(fig:causalvar), we show that, as expected, the average number of causal rare variants and the average probability of disease is greater for the cases with family history, followed by cases without family history, and lastly for controls. This simulated dataset comprises of 1000 controls, 1000 cases without family history, and 1000 cases with family history for three levels of conditional recurrence risk ratios (columns: $\lambda=2,4,8$) and 1 or 2 siblings as family history (rows).

```{r, include=FALSE}
##knitr::include_graphics('figures/causalvar2.png')

idat = read.table("../results/variant-stats.csv", header=TRUE, sep=",")

# n causal variants
idat$prev_target <- factor(idat$prev_target)
plotdat <- idat[idat$measure%in%"n.causal.IP",]

p1 = ggplot(data=plotdat,aes(x=group,y=mean,fill=prev_target))+
  geom_bar(color="black",stat="identity",position = "dodge")+ggtitle("Number of causal rare variants")+
  facet_grid(Nsibsize~lambda)+labs(y="Average (of 1000 Simulations)",x="Group")+
  guides(fill = guide_legend("Target Prev"))+
  geom_errorbar(aes(ymin=mean, ymax=mean+sd), width=.2,
                position=position_dodge(.9))+ 
  geom_text(
    aes(label = round(mean,2), y = mean+sd+0.08),
    position = position_dodge(0.9),
    vjust = 0, size=2
  ) + guides(fill=FALSE) +
    theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ) + ylim(low=0, high=0.67)

# prob disease
idat$prev_target <- factor(idat$prev_target)
plotdat <- idat[idat$measure%in%"pdis.IP",]
p2 = ggplot(data=plotdat,aes(x=group,y=mean,fill=prev_target))+
  geom_bar(color="black",stat="identity",position = "dodge")+ggtitle("Probability of disease")+
  facet_grid(Nsibsize~lambda)+labs(y="Average (of 1000 Simulations)",x="Group")+
 # theme(axis.text.x=element_blank(),axis.ticks.x=element_blank(), axis.title.x=element_blank())+
  guides(fill = guide_legend("Target Prev"))+
  geom_errorbar(aes(ymin=mean, ymax=mean+sd), width=.2,
                position=position_dodge(.9))+ 
  geom_text(
    aes(label = round(mean,2), y = mean+sd+0.015),
    position = position_dodge(0.9),
    vjust = 0, size=2
  ) + ##guides(fill=FALSE) +
      theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        legend.justification="left",
        axis.line = element_line(colour = "grey")##,
        ) + ylim(low=0, high=0.17)


g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
leg = g_legend(p2)

p2noleg = p2 + guides(fill=FALSE)
```


```{r causalvar,fig.align='center', fig.cap='Average of 1000 simulations of number of causal rare variants (left) and probability of disease (right) in proband for three groups: controls, cases without family history, and cases with family history under two disease prevalences (red=0.01, blue=0.05), with one (top) or two (bottom) siblings, and three conditional recurrence risk ratios as columns.', fig.height=4, fig.width=10}
grid.arrange(p1,p2noleg,leg,ncol=3,widths=c(4,4,1))
```




## Type I Error Simulations

Figure \@ref(fig:qq) shows the quantile-quantile (QQ) plots of 10,000 null simulations with 750 subjects per group (2250 in total), target disease prevalence of $0.01$, and $\lambda=2$ (other simulation designs showed similar results and are listed in the \missing{supplement}). We show the comparison with ordinal GAMuT, SKAT and burden test. All methods compared properly control the type I error.


```{r qq, out.width='33%', fig.cap="Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} and the ordinal GAMuT here proposed, with target disease prevalence of 0.01, lambda=2. Simulated datasets (100,000) assumed a 10kb region and rare variants defined as those with MAF <3\\%.",fig.show='hold'}
knitr::include_graphics(c('figures/qq-burden.pdf','figures/qq-skat.pdf','figures/qq-gamut.pdf'))
```

## Power Simulations

Now, we compare the power of ordinal GAMuT with SKAT and burden test (figure \@ref(fig:power-plot)). The power was estimated by computing the proportion of p-values less than the significance level ($\alpha=0.05$ for effect size of 2, and $\alpha=5 \times 10^{-5}$ for effect sizes of 4 and 6) out of 1000 replicates per scenario and model.
For these power simulations, we use different effect sizes in figures ($C=2,4,6$). Columns refer to the conditional recurrence risk ratio $\lambda=2,4,8$, and rows refer to disease prevalences $0.01,0.05$ . We present the scenario where family history involves parents only, no siblings, and we compare the empirical power for sample sizes of $N=400,750,1000,1500$ subjects per group. Our method (ordinal GAMuT) outperformed the burden test [@Li2008] and SKAT [@Wu2011], with power increasing as sample size increases and as rare variant effect size increase.


```{r, include=FALSE}

## RV effect size = 2
dat = read.table("../results/target-prevalence-rv2.csv", header=TRUE)
dat$method = factor(dat$method, levels=c("Ordinal GAMuT", "SKAT", "Burden Test"))
alpha <- 5e-2
dat=within(dat,prev_target<-factor(prev_target, labels=c("Prevalence 0.01", "Prevalence 0.05")))
dat2 = subset(dat,Nsibsize=="Parents Only")
p1 = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  ##guides(color=guide_legend(title="Method")) +
  guides(color=FALSE) +
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 2")

## RV effect size = 4
dat = read.table("../results/target-prevalence-rv4.csv", header=TRUE)
dat$method = factor(dat$method, levels=c("Ordinal GAMuT", "SKAT", "Burden Test"))
alpha <- 5e-5
dat=within(dat,prev_target<-factor(prev_target, labels=c("Prevalence 0.01", "Prevalence 0.05")))
dat2 = subset(dat,Nsibsize=="Parents Only")
p2 = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  ##guides(color=guide_legend(title="Method")) +
  guides(color=FALSE)+
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 4")

## RV effect size = 6
dat = read.table("../results/target-prevalence-rv6.csv", header=TRUE)
dat$method = factor(dat$method, levels=c("Ordinal GAMuT", "SKAT", "Burden Test"))
alpha <- 5e-5
dat=within(dat,prev_target<-factor(prev_target, labels=c("Prevalence 0.01", "Prevalence 0.05")))
dat2 = subset(dat,Nsibsize=="Parents Only")
p3 = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")
        ##legend.justification="left"
        ##legend.position="bottom"
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  guides(color=guide_legend(title="Method")) +
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 6")

# Extract the plot legend as a separate grob
# http://stackoverflow.com/questions/12539348/ggplot-separate-legend-and-plot
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
leg = g_legend(p3)

p3no.leg = ggplot(data = dat2, aes_string(x = "n_per_group", y = "power", color="method")) + 
  geom_line() + geom_point(alpha=0.6) + 
  scale_x_continuous(labels = unique(dat2[,"n_per_group"]), breaks = unique(dat2[,"n_per_group"])) +
  theme(
        plot.title = element_text(hjust=0.5, size=rel(1.2)),
        axis.title.x = element_text(size=rel(1.2)),
        axis.title.y = element_text(size=rel(1.2), angle=90, vjust=0.5, hjust=0.5),
        axis.text.x = element_text(colour="grey", size=rel(1.4), angle=90, hjust=.5, vjust=.5, face="plain"),
        axis.text.y = element_text(colour="grey", size=rel(1.4), angle=0, hjust=.5, vjust=.5, face="plain"),
        panel.background = element_rect(fill = NA, color = "black"),
        axis.line = element_line(colour = "grey")##,
        ##legend.position="bottom"
        ) +
  scale_y_continuous(limits = c(0, 1), expand = c(0,0)) + 
  facet_grid(paste0("prev_target","~","lambda")) + 
  guides(color=FALSE) +
  xlab("Samples/Group") + 
  ylab(bquote(.('Empirical Power:')~alpha == .(alpha)))+
  ggtitle("RV effect size = 6")
```


```{r power-plot,fig.align='center', fig.cap='Power for gene-based testing comparing three methods: burden test (red), SKAT (green) and ordinal GAMuT (blue). We compared two disease prevalences 0.01, 0.05 (rows), different conditional recurrence risk ratio $\\lambda=2,4,8$ (columns).', fig.height=8, fig.width=7}
##p1
##p2
##p3

##grid.newpage()
##grid.draw(ggarrange(p1,p2,p3, ncol = 1))
##ggarrange(p1,p2,p3, ncol = 1)

##grid.arrange(grobs = lapply(
## list(p1, p2, p3no.leg),
## set_panel_size,
## width = unit(0.75, "in"),
## height = unit(0.99, "in")
## ),leg, ncol=2)

p11 = set_panel_size(p1, width = unit(0.8, "in"), height = unit(1, "in"))
p22 = set_panel_size( p2, width = unit(0.8, "in"), height = unit(1, "in"))
p33 = set_panel_size( p3no.leg, width = unit(0.8, "in"), height = unit(1, "in"))

grid.arrange(p11,p22,p33,leg,ncol=2)
```

## Analysis of Pittsburgh Orofacial Cleft Multiethnic GWAS

We applied our method to a Pittsburgh Orofacial Cleft (POFC) Multiethnic GWAS [@Leslie2016], [@Leslie2016b] with 1,411 Caucasian subjects (267 cases with family history, 309 cases without family history and 835 controls) and 61,671 variants used for annotation with Bystro [@Kotlar2018]. 
We filtered rare variants with MAF $[0.001,0.05]$, and filtered genes to having minimum 4 rare variants, which resulted in 5,137 gene tests. 
We tested the association between the 5,137 genes and CL or CL/P status, adjusting for PCs for population structure. We compared our results (ordinal GAMuT) with burden test and SKAT approach. 
Neither of the methods show any p-value inflation (Fig. \@ref(fig:cleft-qq)).

None of the methods identified any genes significantly associated with CL/P. However, ordinal GAMuT 
identified one gene (GRHL2) on chromosome 8 that passes the suggestive significance threshold (Fig. \@ref(fig:cleft-manhattan)). GRHL2 is in the same gene family as GRHL3, which is a transcription factor that is mutated in syndromic forms of clefting and is associated with nonsyndromic clefting [@Leslie2016], [@Leslie2016b], [@Carpinelli2017], [@PeyrardJanvid2014].


```{r, include=FALSE}
load("../original-cleft/analysis_cleft_pass_9/by_gene/res_with_loc.Rda")
colnames(res_with_loc) <- c("gene","p.burden","p.skat","p.gamut","chr","bp")

## significance thresholds
sugg = -log10(1e-04) ##=4
gen = -log10(0.05/nrow(res_with_loc)) ##= 5.01174 
```

```{r cleft-qq, fig.width=12, fig.height=4, fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} and the ordinal GAMuT here proposed.',fig.show='hold'}
layout(matrix(c(1,2,3),nrow=1))
qq(res_with_loc$p.gamut, main = "Ordinal GAMuT")
qq(res_with_loc$p.skat, main = "SKAT")
qq(res_with_loc$p.burden, main = "Burden")
```

<!--
```{r cleft-qq2, out.width='33%', fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for three methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} and the ordinal GAMuT here proposed.',fig.show='hold'}
knitr::include_graphics(c('figures/cleft-burden-qq.png','figures/cleft-skat-qq.png','figures/cleft-ordinal-gamut-qq.png'))
```
-->



```{r cleft-manhattan, fig.width=12, fig.height=4, fig.cap='Gene-based test on Pittsburgh Orofacial Cleft (POFC) Multiethnic GWAS using burden, SKAT and ordinal GAMuT approach. Manhattan plots for each of the three tests. Red line: genome-wide significance level ($-\\log_{10}(0.05/5137) = 5.0117$). Blue line: suggestive level ($-\\log_{10}(1 \\times 10 ^{-4}) = 4$).',fig.show='hold'}
layout(matrix(c(1,2,3),nrow=1))
manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.gamut",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL, annotatePval = 1e-3,
  suggestiveline = -log10(1e-04), genomewideline = -log10(0.05/5137),main="Ordinal GAMuT",ylim=c(0,6))

manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.skat",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL, annotatePval = 1e-3,
  suggestiveline = sugg, genomewideline = gen, main="SKAT",ylim=c(0,6))

manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.burden",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL, annotatePval = 1e-3,
  suggestiveline = sugg, genomewideline = gen, main="Burden",ylim=c(0,6))
```

<!--
```{r cleft-manhattan2, out.width='33%', fig.cap='Gene-based test on Pittsburgh Orofacial Cleft (POFC) Multiethnic GWAS using burden, SKAT and ordinal GAMuT approach. Manhattan plots for each of the three tests. Red line: genome-wide significance level ($-\\log_{10}(0.05/5137) = 5.0117$). Blue line: suggestive level ($-\\log_{10}(1e-04) = 4$).',fig.show='hold'}
knitr::include_graphics(c('figures/cleft-burden.png','figures/cleft-skat.png','figures/cleft-ordinal-gamut.png'))
```
-->

### Replication analysis

Using the dataset in @Mostowska2018, we tried to replicate the top genetic associations for CL/P status. As mentioned already, the replication dataset only included the region 8q22.3 (hg19 chr8:$101,600,001-106,200,000$) and 845 individuals (570 were controls and 275 were affected with nsCL/P). SNPs with SNP call rate less than $0.95$ were excluded as well as samples with sample call rate less than $0.95$. For this dataset, we do not have family history information (and thus, we cannot fit the ordinal GAMuT method here proposed). Instead, the phenotype used was case-control status on CL/P.

We first ran a standard GWAS analysis with plink for the common variants ($MAF>0.05$) under the additive model, but none of the common variants were identified as significantly associated with CL/P affected status. The variant `rs965660` was the top hit, with adjusted p-value of $0.064419$ (at significance level of $5\%$). We used the methodology in @Conneely2007 to adjust the p-values of correlated tests.
The Q-Q plot of the unadjusted p-values is shown in figure \@ref(fig:qqplot-repl).

```{r qqplot-repl, fig.width=3, fig.height=3, fig.cap='Q-Q plots of p-values for association tests of common variants for the replication dataset. The variants belong to the region in chromosome 8 where the ordinal GAMuT top genes were found.',fig.show='hold'}
dat = read.table("../replication/data/data_us/plink-sex.assoc.logistic", header=TRUE)
dat2 = subset(dat,TEST=="ADD") ##removing pvalues for SEX
qq(dat2$P)
```


Regarding the gene-based tests for rare variants, we fit the burden test [@Li2008] and the SKAT test [@Wu2011] on 136 annotated variants with Bystro [@Kotlar2018]. We used variants with MAF between $0.1\%$ and $5\%$, which were annotated in 8 genes.
None of these genes were found to be genome-wide significant, but YWHAZ, NCALD and RIMS2 were above the suggestive significance threshold of $0.05$. Gene GRHL2 had a p-value of 0.132 with the burden test and 0.581 with the SKAT test.

<!--
``````{r repl-manhattan, fig.width=8, fig.height=4, fig.cap='Gene-based test on the replication dataset [@Mostowska2018] using burden, and SKAT approach. Manhattan plots for each of the two tests. Red line: genome-wide significance level ($-\\log_{10}(0.05/8) = 2.20412$). Blue line: suggestive level ($-\\log_{10}(0.05) = 1.301$).',fig.show='hold'}
load("../replication/data/data_us/by_gene/res_with_loc.Rda")
colnames(res_with_loc) <- c("gene","p.burden","p.skat","chr","bp")

## significance thresholds
sugg = -log10(0.05) ##=1.301
gen = -log10(0.05/nrow(res_with_loc)) ##= 2.20412 

layout(matrix(c(1,2),nrow=1))
manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.skat",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL,annotatePval = 0.06, annotateTop = FALSE,
  suggestiveline = sugg, genomewideline = gen, main="SKAT",ylim=c(0,6), xlim=c(9e7, 1.2e8))
manhattan(res_with_loc, chr = "chr", bp = "bp", p = "p.burden",snp="gene",
  col = c("blue4", "orange3"), chrlabs = NULL,annotatePval = 0.06, annotateTop = FALSE,
  suggestiveline = sugg, genomewideline = gen, main="Burden",ylim=c(0,6), xlim=c(9e7, 1.2e8))
```


```{r repl-qq, fig.width=3.5, fig.height=3.5, fig.cap='Q-Q plots of p-values for gene-based tests of rare variants for two methods: burden test \\citep{Li2008}, SKAT \\citep{Wu2011} on xxx dataset.',fig.show='hold', fig.align="center"}
load("../replication/data/data_us/by_gene/res_with_loc.Rda")
colnames(res_with_loc) <- c("gene","p.burden","p.skat","chr","bp")
source("../scripts/qqplot-MF.r")

layout(matrix(c(1,2),nrow=1))
qqunif.plot(res_with_loc$p.skat, main="SKAT")
qqunif.plot(res_with_loc$p.burden, main = "Burden")
```
-->

# Discussion

Standard methods for case-control association studies of rare and
common variation often treat disease outcome as a dichotomous
phenotype. However, both theoretical and experimental studies have
demonstrated that cases with a family history of disease can be
enriched for risk variation relative to sporadic cases. Likewise,
controls with a family history of disease can harbor more risk
variation than controls without such history.
Assuming family-history information is available, this observation
motivates the idea of replacing the standard dichotomous outcome
variable in case-control studies with a more informative outcome
variable that distinguishes cases with and without family history, and
controls. The rationale is that subjects with family history can
be more enriched for variants of most common diseases. That is, we
expect to see more causal variants in cases with family history and
want to leverage this trend when exploring the genetic contributors of complex traits.
Here, we propose an extension to the cross-phenotype association method GAMuT [@Broadaway2016] 
which tests association among ordinal phenotype data with high-dimensional genotype data. 

\missing{xxxxxxxxx}

We introduced an extension to the GAMuT method [@Broadaway2016] to incorporate family information to enhance case-control association studies. This approach converts the usual binary phenotype of case-control status into an ordinal phenotype with three levels: cases with family history, cases without family history and controls, and it allows to adjust for covariates. 
Even though we do not include controls with family history, this ordinal approach can easily be extended to the case of four categories: controls with and without family history, and cases 
with and without family history by considering an ordinal phenotype with 4 levels.
Finally, just as the standard GAMuT test, the ordinal GAMuT obtains analytic p-values from Davies' method [@Davies1980] which is computationally efficient, allowing the analysis of datasets in the genomic scale.

Simulation studies of rare variant sets showed that our ordinal GAMuT method is more powerful compared to usual gene-based tests like burden test [@Li2008] and SKAT [@Wu2011]. 
Applying our method to Pittsburgh Orofacial Cleft Multiethnic GWAS [@Leslie2016], [@Leslie2016b], we identified a gene (GRHL2) (not previously reported) to suggestively associate with cleft lip and palate phenotypes. 
GRHL2 is in the same gene family as GRHL3, which is a transcription factor that is mutated in syndromic forms of clefting and is associated with nonsyndromic clefting [@Leslie2016], [@Leslie2016b], [@Carpinelli2017], [@PeyrardJanvid2014].
Burden and SKAT on these same phenotypes (figure \@ref(fig:cleft-manhattan)) failed to identify any significant or suggestive genes.

The extension of GAMuT to include family history information could lead to future lines of work. In particular, potential options for enhanced outcome variables could involve conditional means from liability-threshold models. \missing{(expand, maybe also mention you can do the same kind of analysis for gene-based analysis of common variation?)}

See section of Web Resources for the R scripts to run ordinal GAMuT.



**Acknowledgements:** Samples provided by Kaare Christensen (University of Southern Denmark), Frederic W.B. Deleyiannis (University of Colorado School of Medicine, Denver), Jacqueline T. Hecht (McGovern Medical School and School of Dentistry UT Health at Houston), George L. Wehby (University of Iowa), Seth M. Weinberg (University of Pittsburgh), Jeffrey C. Murray (University of Iowa) and Mary L. Marazita (University of Pittsburgh).
This work was supported by NIH grants GM117946 [MP,DG], HG007508 [MP], R00-DE025060 [EJL], X01-HG007485 [MLM], R01-DE016148 [MLM, SMW], U01-DE024425 [MLM], R37-DE008559 [JCM, MLM], R21-DE016930 [MLM], R01-DE012472 [MLM], R01-DE011931 [JTH], R01-DE011948 [KC], U01-DD000295 [GLW]; NIH contract to the Johns Hopkins Center for Inherited Disease Research: HHSN268201200008I.


**Web Resources: ** The URLs for software: \url{https://github.com/crsl4/ordinal-gamut} and \url{http://www.genetics.emory.edu/labs/epstein/software}.

<!--
To put appendix after references: https://stackoverflow.com/questions/41532707/include-rmd-appendix-after-references
-->

<!--
## Unused code, maybe useful for some day
# g1 = ggplotGrob(p1)
# g2 = ggplotGrob(p2)
# g3 = ggplotGrob(p3)
# grid::grid.newpage()
# ##grid::grid.draw(rbind(g1,g2, g3))
# ##gridExtra::grid.arrange(g1,g2,g3)
# library(ggpubr)
# par(mar=c(0,0,0,0))
# ggarrange(g1,g2,g3,heights = c(3,3,3),
#           ncol = 1, nrow = 3, align = "v")


# Extract the plot legend as a separate grob
# http://stackoverflow.com/questions/12539348/ggplot-separate-legend-and-plot
g_legend<-function(a.gplot){
  tmp <- ggplot_gtable(ggplot_build(a.gplot))
  leg <- which(sapply(tmp$grobs, function(x) x$name) == "guide-box")
  legend <- tmp$grobs[[leg]]
  legend
}
thm = theme(legend.justification="left")
widths = c(8,5)
leg = g_legend(p1)

library(gridExtra)
library(ggpubr)
library(grid)
grid.arrange(p1 + guides(color=FALSE), nullGrob(), widths=widths)
grid.arrange(p2 + guides(color=FALSE), leg, widths=widths)
grid.arrange(p2 + guides(color=FALSE), nullGrob(), widths=widths)
-->